### PR TITLE
Topic4

### PR DESCRIPTION
**Pull request by jaiup**

### CODE REVIEW

**Review by afiorenza — COMMENTED** [review]
Everything okey. But i see that you are using strings like "comment" and "commentIndex" several times. Could you move them to constants in order to make this more maintainable? Thanks!

File: 04-HTML5-APIs/js/web-socket.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+function doSend(message) {
+  writeToScreen("SENT: " + message);
+  websocket.send(message);
+}
+
+function writeToScreen(message) {
+  let pre = document.createElement("p");
+  pre.style.wordWrap = "break-word";
```
Comment (afiorenza) [inline 04-HTML5-APIs/js/web-socket.js]: It is better to have the style in a css stylesheet.

File: 04-HTML5-APIs/js/web-socket.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  writeToScreen('<span style="color: red;">ERROR:</span> ' + evt.data);
+}
+
+function doSend(message) {
+  writeToScreen("SENT: " + message);
+  websocket.send(message);
+}
+
+function writeToScreen(message) {
+  let pre = document.createElement("p");
```
Comment (afiorenza) [inline 04-HTML5-APIs/js/web-socket.js]: I think `pre` is not a descriptive name. Could you find a better name?

File: 04-HTML5-APIs/js/web-socket.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+function onOpen(evt) {
+  writeToScreen("CONNECTED");
+}
+
+function onClose(evt) {
+  writeToScreen("DISCONNECTED");
+}
+
+function onMessage(evt) {
+  writeToScreen('<span style="color: blue;">RESPONSE: ' + evt.data + '</span>');
```
Comment (afiorenza) [inline 04-HTML5-APIs/js/web-socket.js]: Style should go in the css and you should set a class to this span.

File: 04-HTML5-APIs/js/drag-and-drop.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+let drop = document.getElementById("drop");
+
+drop.ondragenter = function(event) {
+  document.getElementById("drop").textContent = '';
```
Comment (afiorenza) [inline 04-HTML5-APIs/js/drag-and-drop.js]: You already declare drop (`document.getElementById("drop")`) above. You can reuse it.

**Review by afiorenza — CHANGES_REQUESTED** [review]
Well done! just a few changes

File: 04-HTML5-APIs/js/web-socket.js (hunk 3 above)
Comment (jaiup) [reply 04-HTML5-APIs/js/web-socket.js]: ok

File: 04-HTML5-APIs/js/web-socket.js (hunk 2 above)
Comment (jaiup) [reply 04-HTML5-APIs/js/web-socket.js]: ok

File: 04-HTML5-APIs/js/web-socket.js (hunk 1 above)
Comment (jaiup) [reply 04-HTML5-APIs/js/web-socket.js]: ok

File: 04-HTML5-APIs/js/drag-and-drop.js (hunk 4 above)
Comment (jaiup) [reply 04-HTML5-APIs/js/drag-and-drop.js]: ok

**Review by afiorenza — APPROVED** [review]

**Pull request merged**